ui/lib/sync/addon/components/secrets/page/destinations/destination/sync.hbs

```diff
@@ -13,61 +13,75 @@
-        data-test-sync-mount-select
-      />
-    {{else}}
-      <FormFieldLabel @label="Enter an existing KV engine mount" />
-      <FilterInput
-        placeholder="KV engine mount path"
-        value={{this.mountPath}}
-        @onInput={{fn (mut this.mountPath)}}
-        data-test-sync-mount-input
-      />
-    {{/if}}
 
-    <KvSuggestionInput
-      @label="Select a secret to sync"
-      @subText="Enter the full path to the secret. Suggestions will display below if permitted by policy."
-      @value={{this.secretPath}}
-      @mountPath={{this.mountPath}}
-      @onChange={{fn (mut this.secretPath)}}
-    />
+    <p class="is-label">Which secrets would you like us to sync?</p>
+    <p class="sub-text">
+      Select a KV engine mount and path to sync a secret to the destination.
+    </p>
 
-    <div class="field box is-fullwidth is-bottomless has-top-margin-xxxl">
-      <Hds::ButtonSet>
-        <Hds::Button
-          @text="Sync to destination"
-          @color="primary"
-          @icon={{if this.setAssociation.isRunning "loading"}}
-          type="submit"
-          disabled={{or (not (and this.mountPath this.secretPath)) this.isSecretDirectory this.setAssociation.isRunning}}
-          data-test-sync-submit
+    <div class="has-top-margin-l">
+      {{#if this.mounts}}
+        <SearchSelect
+          @label="Select a mount for the KV engine"
+          @options={{this.mounts}}
+          @selectLimit={{1}}
+          @disallowNewItems={{true}}
+          @onChange={{this.setMount}}
+          data-test-sync-mount-select
         />
-        <Hds::Button @text="Back" @color="secondary" {{on "click" this.back}} data-test-sync-cancel />
-      </Hds::ButtonSet>
-      {{#if this.isSecretDirectory}}
-        <AlertInline
-          @type="warning"
-          @message="Syncing secret directories is not available at this time, please type a path to a single secret"
-          class="has-top-margin-s"
+      {{else}}
+        <FormFieldLabel @label="Enter an existing KV engine mount" />
+        <FilterInput
+          placeholder="KV engine mount path"
+          value={{this.mountPath}}
+          @onInput={{fn (mut this.mountPath)}}
+          data-test-sync-mount-input
         />
       {{/if}}
+
+      <KvSuggestionInput
+        @label="Select a secret to sync"
+        @subText="Enter the full path to the secret. Suggestions will display below if permitted by policy."
+        @value={{this.secretPath}}
+        @mountPath={{this.mountPath}}
+        @onChange={{fn (mut this.secretPath)}}
+      />
+
+      <div class="field box is-fullwidth is-bottomless has-top-margin-xxxl">
+        <Hds::ButtonSet>
+          <Hds::Button
+            @text="Sync to destination"
+            @color="primary"
+            @icon={{if this.setAssociation.isRunning "loading"}}
+            type="submit"
+            disabled={{or (not (and this.mountPath this.secretPath)) this.isSecretDirectory this.setAssociation.isRunning}}
```

Comment:
This logic is a bit 😵  -- could we move it to a getter? at least all of `or (not (and this.mountPath this.secretPath)) this.isSecretDirectory`

Reply:
Yes good call! I hadn't noticed this get piled on over time 😁.